mne/viz/evoked.py

```diff
@@ -1200,3 +1200,439 @@ def plot_evoked_joint(evoked, times="peaks", title='', picks=None,
+        picked_chans = []
+        pairpicks = _pair_grad_sensors(example.info, topomap_coords=False)
+        for ii in np.arange(0, len(pairpicks), 2):
+            first, second = pairpicks[ii], pairpicks[ii + 1]
+            if first in picks or second in picks:
+                picked_chans.append(first)
+                picked_chans.append(second)
+        picks = list(sorted(set(picked_chans)))
+        ch_names = [example.ch_names[pick] for pick in picks]
+
+    if ymin is None and (gfp is True or ch_type == 'grad'):
+        ymin = 0  # 'grad' and GFP are plotted as all-positive
+
+    # deal with dict/list of lists and the CI
+    if not isinstance(ci, np.float):
+        msg = '"ci" must be float, got {0} instead.'
+        raise TypeError(msg.format(type(ci)))
+
+    # if we have a dict/list of lists, we compute the grand average and the CI
+    if not all([isinstance(evoked_, Evoked) for evoked_ in evokeds.values()]):
+        if ci is not None and gfp is not True:
+            # calculate the CI
+            sem_array = {}
+            for condition in conditions:
+                # this will fail if evokeds do not have the same structure
+                # (e.g. channel count)
+                if ch_type == 'grad' and gfp is not True:
+                    data = np.asarray([
+                        _merge_grad_data(
+                            evoked_.data[picks, :]).mean(0)
+                        for evoked_ in evokeds[condition]])
+                else:
+                    data = np.asarray([evoked_.data[picks, :].mean(0)
+                                       for evoked_ in evokeds[condition]])
+                sem_array[condition] = _ci(data, ci)
+
+        # get the grand mean
+        evokeds = dict((cond, combine_evoked(evokeds[cond], weights='equal'))
+                       for cond in conditions)
+
+        if gfp is True and ci is not None:
+            warn("Confidence Interval not drawn when plotting GFP.")
+    else:
+        ci = False
+        combine_evoked(list(evokeds.values()))  # check if they are compatible
+    # we now have dicts for data ('evokeds' - grand averaged Evoked's)
+    # and the CI ('sem_array') with cond name labels
+
+    # let's plot!
+    if axes is None:
+        fig, axes = plt.subplots(1, 1)
+        fig.set_size_inches(8, 6)
+    else:
+        fig = axes.figure
+
+    # style the individual condition time series
+    # first, check if input is valid
+    tags = set([tag for cond in conditions for tag in cond.split("/")])
+    msg = ("Can't map between conditions and the provided {0}. Make sure "
+           "you have provided keys in the format of '/'-separated tags, "
+           "and that these correspond to '/'-separated tags for the condition "
+           "names (e.g., conditions like 'Visual/Right', and styles like "
+           "'colors=dict(Visual='red'))'. The offending tag was '{1}'.")
+    for name_, item in zip(("colors", "linestyles"), (colors, linestyles)):
```

Comment:
this looks similar to what is done in `_setup_styles` - maybe it should be done there, especially if it duplicates some of that functionality.